### DISCUSSION TITLE
Procedure for adding details on new website needs clarification.

### DISCUSSION THREAD

**Discussion by Budgie2** [keepassxreboot/keepassxc#6867] [Q&A] [answered]
I have a problem, possibly through my ignorance, with registering on a website in a reliable and secure way for the first time.

The background situation is that I have keepassxc installed, open and sitting minimised in my tray.

Using my browser, if I go to a website for the first time and wish to register, the usual procedure is to enter my password and then I am asked for a password.  At this point I do not have a pw but can get one from keepassxc.  

If I then copy and paste this newly generated pw into the registration field and proceed, at this point I have no knowledge or sight of the password and the pw is not saved anywhere other than in the registration on the remote website.
  
If I subsequently log in to the new website, once registered, I need the new pw but where is it?  I have not at that stage created a new entry into the database.  I may have been doing this the following day having waited for a confirmation email so unless I have written it done or otherwise saved it, I am stuck.

OTOH if I create a new entry in advance of registering I can use it to generate and also save a pw but when I do log in, keepassxc asks me automatically to create a new entry.  I can at this stage select which subdirectory the new entry is placed but I have no ability to give the new entry a Title or select the new one I had previously created for this website. 

Sorry to be so dumb but what is correct procedure and am I missing something?

**Comment by phoerious** [comment, marked as answer]
The temporary password generator is for throwaway passwords that you don't permanently. Don't use that for new accounts.

The proper workflow is: create a new entry in KeePassXC with a new random password and save it. Then go to your online service and create the account with the password copied from the new entry.

**Reply by Budgie2** [reply]
Hi and thanks for the early reply.  Glad that what I have been doing is the right approach but I am having difficulty getting the right login url back into the data.  Will persevere.  
A related problem which has provoked my original post is the confusion resulting from having several apps hosted on github or GNOME-github.   Should I just use the one github login and then feel my way to required app or set up different logins for each app.  I would prefer just having the one.  Is that the right way?

**Reply by phoerious** [reply]
Just select the URL of the login page in your browser and copy-paste it into the entry.
I'm not quite sure what you mean by "several apps". Do you have multiple logins on GitHub?

**Reply by Budgie2** [reply]
It seems I have and am having all kinds of problems.  I need to do some house keeping but I have github and gitlab and the passwords are scrambled so before I delete everything, I would very much appreciate your help.  I make occasional use of github for various apps and have recently raised issues on keepassxc, puddletag and easytag.  This is where I became unstuck.

**Reply by phoerious** [reply]
I cannot help you with "various issues". If you have multiple logins for the same page, you can create multiple entries with the same URL. You will get a selection of all matching entries in the browser extension.
